app/kuma-dp/pkg/dataplane/metrics/server.go

```diff
@@ -153,28 +170,35 @@ func rewriteMetricsURL(address string, port uint32, path string, queryModifier Q
 func (s *Hijacker) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
 	ctx := req.Context()
 	out := make(chan []byte, len(s.applicationsToScrape))
+	contentTypes := make(chan expfmt.Format, len(s.applicationsToScrape))
```

Comment:
nit: I'd prefer 1 channel with results instead of 2 channels, something like:
```go
type content struct {
    type expfmt.Format
    data []byte
}

func (s *Hijacker) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
...
    out := make(chan []content, len(s.applicationsToScrape))
...
}
```

But that's really up to you, should not be a blocker for this PR

Reply:
Yes, I initially thought of implementing something similar. But as we have to set the headers before writing any data to the response body, we have to call `selectContentType()` as we are doing now before writing data to response body. For this we have to pass the `[]contents` to the function and to avoid unnecessary copy of contents to the function, we have to pass pointer to the []contents. So to kind of avoid these things I simply preferred using two channels.